OpenRA.Mods.Common/UtilityCommands/PngSheetImportMetadataCommand.cs

```diff
@@ -30,7 +32,23 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 			using (var pngStream = File.OpenRead(args[1]))
 				png = new Png(pngStream);
 
-			foreach (var node in MiniYaml.FromFile(Path.ChangeExtension(args[1], "yaml")))
+			var yaml = MiniYaml.FromFile(Path.ChangeExtension(args[1], "yaml"));
+
+			var frameSizeField = yaml.Where(y => y.Key == "FrameSize").Select(y => y.Value.Value).FirstOrDefault();
+			if (frameSizeField != null)
+			{
+				var frameSize = FieldLoader.GetValue<Size>("FrameSize", frameSizeField);
+
+				var frameAmountField = yaml.Where(y => y.Key == "FrameAmount").Select(y => y.Value.Value).FirstOrDefault();
+				if (frameAmountField != null)
+				{
+					var frameAmount = FieldLoader.GetValue<int>("FrameAmount", frameAmountField);
+					if (frameSize.Width * frameSize.Height * frameAmount > png.Width * png.Height)
```

Comment:
This fails to catch any case where `(png.Width / frameSize.Width) * (png.Height % frameSize.Height) + (png.Width % frameSize.Width) * (png.Height / frameSize.Height) + (png.Width % frameSize.Width) * (png.Height % frameSize.Height) >= frameSize.Width * frameSize.Height` or in simple terms, when the unusable margin has an area at least as big as one frame. For example a 24 x 32 image has the area of 3 16 x 16 frames but can only hold two.
Calculating the maximum number of frames an image can hold is simple enough:

```suggestion
					if (frameAmount > (png.Width / frameSize.Width) * (png.Height / frameSize.Height))
```

Reply:
Thanks.